src/tiled/gotodialog.h

```diff
@@ -0,0 +1,42 @@
+#ifndef GOTODIALOG_H
+#define GOTODIALOG_H
+
+#include <QApplication>
+#include <QDialog>
+#include <QGroupBox>
+#include <QHBoxLayout>
+#include <QIntValidator>
+#include <QLabel>
+#include <QLineEdit>
+#include <QPushButton>
+#include <QString>
+
+namespace Tiled {
+
+namespace Internal {
+
+class GotoDialog : public QDialog
+{
+    Q_OBJECT
+
+public:
+    GotoDialog(QWidget *parent = nullptr, Qt::WindowFlags flags = Qt::WindowFlags());
+    static GotoDialog *instance() { return mInstance; }
+    static GotoDialog *showDialog();
+
+signals:
+    void changeCoordinates(int x, int y);
+
+private:
+    void goToCoordinates();
+    QLineEdit *lineEditX;
+    QLineEdit *lineEditY;
+
+    static GotoDialog *mInstance;
+};
+
+}
```

Comment:
Coding style: because it can be hard to see what these closing brackets are for, I have a habit of annotating them as follows:
```c++
} // namespace Internal
} // namespace Tiled
```